### PR TITLE
Allow disabling Slither

### LINKED ISSUE #1318: Option to explicitly disable slither
### Describe the desired feature

Disabling slither speeds up the startup time considerably. It would be useful to have an option to disable slither during development of the fuzzing suite.

### PR DESCRIPTION
**Pull request by elopez**
This allows explicitly disabling Slither with a new `--disable-slither` command-line flag or `disableSlither` configuration option. A blinking, red warning will be shown in the UI when Slither is disabled or otherwise failed, to dissuade people from fuzzing without it.

Closes #1318

### CODE REVIEW

File: lib/Echidna/SourceAnalysis/Slither.hs [hunk 1]
```diff
@@ -177,3 +182,7 @@ runSlither fp solConf = do
 
 emptySlitherInfo :: SlitherInfo
 emptySlitherInfo = SlitherInfo mempty mempty mempty mempty mempty [] [] []
+
+isEmptySlitherInfo :: Maybe SlitherInfo -> Bool
+isEmptySlitherInfo (Just (SlitherInfo _ _ _ _ _ [] [] [])) = True
```
Comment (elopez) [inline lib/Echidna/SourceAnalysis/Slither.hs:187]: `Nothing` should probably also be empty

**Comment by rappie** [comment]
:partying_face:  :tada:

File: lib/Echidna/SourceAnalysis/Slither.hs [hunk 2]
```diff
@@ -145,8 +145,13 @@ instance FromJSON SlitherInfo where
 
 -- Slither processing
 runSlither :: FilePath -> SolConf -> IO SlitherInfo
-runSlither fp solConf = do
-  findExecutable "slither" >>= \case
+runSlither fp solConf = if solConf.disableSlither
+  then do
+    hPutStrLn stderr $
+        "WARNING: slither was expliticly disabled. Echidna uses Slither (https://github.com/crytic/slither)"
```
Comment (elopez) [inline lib/Echidna/SourceAnalysis/Slither.hs:151]:
```suggestion
        "WARNING: Slither was explicitly disabled. Echidna uses Slither (https://github.com/crytic/slither)"
```

**Review by arcz — APPROVED** [review]
lgtm

**Pull request merged**